extensions/wikia/PortableInfobox/services/Parser/MediaWikiParserService.php

```diff
@@ -0,0 +1,40 @@
+<?php
+namespace Wikia\PortableInfobox\Parser;
+
+class MediaWikiParserService implements ExternalParser {
+
+	protected $parser;
+	protected $frame;
+	protected $localParser;
+
+	public function __construct( \Parser $parser, \PPFrame $frame ) {
+		$this->parser = $parser;
+		$this->frame = $frame;
+	}
+
+	public function parse( $wikitext ) {
+		$preprocessed = $this->parser->recursivePreprocess( $wikitext, $this->frame );
```

Comment:
we should extract expanding to other method, and use only if necessary, for performance reasons